### PR TITLE
feat(kfam): support multiple cluster admin

### PR DESCRIPTION
**Pull request by hhk7734**
This PR enables support for multiple cluster admins.

```shell
/access-management -cluster-admin "<userA>,<userB>"
```

Originally, this was implemented in https://github.com/kubeflow/kubeflow/pull/7267, but since the kfam codebase has been moved here, I'm reopening the PR in this repository.

### CODE REVIEW

**Comment by hhk7734** [comment]
ping

**Comment by juliusvonkohout** [comment]
@andyatmiami

**Comment by andyatmiami** [comment]
Hey, happy to take a look at this - but I'm still _getting acquainted_ with all the various pieces of `kubeflow/dashboard`...

Could you help me out by explaining how I could test this to verify?  The code changes themselves are straigthforward/make sense - but I'd like to "see" the outcome..

File: components/access-management/kfam/api_default.go [hunk 1]
```diff
@@ -291,15 +291,15 @@ func (c *KfamV1Alpha1Client) getUserEmail(header http.Header) string {
 }
 
 func (c *KfamV1Alpha1Client) isClusterAdmin(queryUser string) bool {
```
Comment (andyatmiami) [inline components/access-management/kfam/api_default.go:293]: I realize this is _pre-existing code_ not modified in this PR - but upon analyzing the code - I feel slightly uneasy by the fact that a `queryUser` value of `""` (empty string) - would/could actually return `true` here...

This gets potentially more problematic with the changes introduced in this PR - as a leading or trailing `,` could lead to `""` passing an admin check.

While I wholly appreciate its unlikely _in the real world_ to exploit this - I think a "fail fast" check to return `false` in the event `queryUser` is an empty string would be beneficial.

FYI: @juliusvonkohout @kimwnasptd 

This could in theory constitute a _breaking change_ if in some wild circumstance someone was relying on empty string being a sort of "super user" -but reality of that being relied upon seems low.

**Review by andyatmiami — CHANGES_REQUESTED** [review]
While I realize `api_default.go` does **NOT** currently have unit tests - given the subtle edge cases that `strings.Split` can introduce - I think it would be nice to include a simple `api_default_test.go` file that leverages table-driven tests to help validate the behavior being included in this PR (particularly if/when we act on other PR comments I have made)

File: components/access-management/main.go [hunk 2]
```diff
@@ -37,16 +38,16 @@ func main() {
...
 	flag.StringVar(&userIdPrefix, USERIDPREFIX, "accounts.google.com:", "Request header user id common prefix")
-	flag.StringVar(&clusterAdmin, CLUSTERADMIN, "", "cluster admin")
+	flag.StringVar(&clusterAdmins, CLUSTERADMIN, "", "cluster admin")
 	flag.Parse()
 
 	profile.AddToScheme(scheme.Scheme)
 	istioSecurityClient.AddToScheme(scheme.Scheme)
 
-	profileClient, err := kfam.NewKfamClient(userIdHeader, userIdPrefix, clusterAdmin)
+	profileClient, err := kfam.NewKfamClient(userIdHeader, userIdPrefix, strings.Split(clusterAdmins, ","))
```
[the remark this answers is not in the document]
Comment (hhk7734) [reply components/access-management/main.go:50]: https://github.com/kubeflow/dashboard/pull/95/commits/1960d4dd9ac54cabe756183dac792755123bd047
This commit changes it to sanitize the list.

**Comment by andyatmiami** [comment]
Ok, i ended up powering through and getting the successful results I was interested in 💯 

I'm attaching the test script I used for verification - but not implying I want to _support it_ for others - as it:
- was heavily vibe coded
- supports only my specific machine setup (MacOS + podman + kind)

But it does deploy kfam with a variety of `-cluster-admin` arguments to truly verify the behavior introduced in this PR.
[test_multi_admin.sh](https://github.com/user-attachments/files/22124694/test_multi_admin.sh)
- should be placed and ran in root `components/access-management` directory

/ok-to-test
/lgtm
